### DISCUSSION TITLE
`GLArea` has to be the direct child of an `ApplicationWindow`?

### DISCUSSION THREAD

**Discussion by Barugon** [gtk-rs/gtk4-rs#1400] [Q&A] [answered]
I wanted to be able to switch to dark or light mode on the fly, so this required that I change my app to use libadwaita. However, adwaita's `ApplicationWindow` doesn't include window decorations. I had to use a `Box` layout with a `HeaderBar` and a `GLArea` but now the `GLArea` won't draw as a child of a `Box`.

Any suggestions?

**Comment by Barugon** [comment]
Wow, `DrawingArea` also has to be the direct window child.

**Comment by bilelmoussaoui** [comment, marked as answer]
Sounds like you forgot to set vexpand on the widget instead. There is nothing special about being a direct child of the window at all.

**Reply by Barugon** [reply]
> Sounds like you forgot to set vexpand on the widget instead. There is nothing special about being a direct child of the window at all.

Well, now I know. Thanks for the tip.
